From 74f2dda73d803afa20e0850fdfaf7aab25a29b52 Mon Sep 17 00:00:00 2001
From: sri anjaneyulu alapati <srix.anjaneyulu.alapati@intel.com>
Date: Wed, 21 Jun 2023 14:43:33 +0000
Subject: [PATCH] Fix Adaptor scope issue in upgraded connection

Ownership of adaptor was moved to lambda function handling 
upgraded connection.
Due to this, during any error(Example :Insufficient privilege) when
the control goes back to http_connection to send error response,
the adaptor will be invalid leading to crash.
This commit retains the ownership of adaptor until connection
is upgraded.

Signed-off-by: sri anjaneyulu alapati <srix.anjaneyulu.alapati@intel.com>
Upstream-Status: Pending
---
 http/routing.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/http/routing.hpp b/http/routing.hpp
index e72af07a..db045285 100644
--- a/http/routing.hpp
+++ b/http/routing.hpp
@@ -1329,9 +1329,9 @@ class Router
         // appear to work with the std::move on adaptor.
         validatePrivilege(
             req, asyncResp, rule,
-            [&rule, asyncResp, adaptor(std::forward<Adaptor>(adaptor))](
+            [&rule, asyncResp, &adaptor](
                 Request& thisReq) mutable {
-            rule.handleUpgrade(thisReq, asyncResp, std::move(adaptor));
+            rule.handleUpgrade(thisReq, asyncResp, std::forward<Adaptor>(adaptor));
             });
     }
 
-- 
2.17.1

